src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -308,14 +306,19 @@ protected virtual Task SetupRequest (HttpRequestMessage request, HttpURLConnecti
 		/// <param name="connection">HTTPS connection to return socket factory for</param>
 		protected virtual SSLSocketFactory? ConfigureCustomSSLSocketFactory (HttpsURLConnection connection)
 		{
-			return _underlyingHander.ConfigureCustomSSLSocketFactory (connection);
+			return _underlyingHander.ConfigureCustomSSLSocketFactoryInternal (connection);
 		}
 
 		[DynamicDependency (DynamicallyAccessedMemberTypes.PublicParameterlessConstructor, typeof (AndroidMessageHandler))]
 		object GetUnderlyingHandler ()
 		{
 			var fieldName = "_nativeHandler";
-			var baseType = GetType ().BaseType;
+			var type = GetType ();
+			// while (type != typeof (AndroidClientHandler)) {
```

Comment:
When this `while` loops is re-enabled, I think a more "conventional" approach would be to instead do:

```csharp
FieldInfo? field = null;
for (var type = GetType (); type != null; type = type.BaseType) {
    field = type.GetField (fieldName, BindingFlags.Instance | BindingFlags.NonPublic);
    if (field != null)
        break;
}
if (field == null)
    throw new InvalidOperationException (…);
…
```

The benefit to this approach is that it's *conceptually simpler* (yay maintenance), and also means that *if someone knows what they're doing* (possible :rofl:), they can "simply" add a new `_nativeHandler` instance field and things will (more or less) Just Work™.

Reply:
I like this approach, thanks for the suggestion!